include/depthai/depthai_wrapper.hpp

```diff
@@ -0,0 +1,86 @@
+#include "depthai/device.hpp"
+#include "nlohmann/json.hpp"
+#include <opencv2/core/core.hpp>
+#include <opencv2/opencv.hpp>
+#include <string>
+#include <unordered_map>
+
+namespace DepthAI {
+
+// shared ptr for cv::Mat
+using CV_mat_ptr = std::shared_ptr<cv::Mat>;
+// tuple of data packets on which Depthai is publishing the streams
+
+using PacketsTuple = std::tuple<
+    std::list<std::shared_ptr<NNetPacket>>,
+    std::list<std::shared_ptr<HostDataPacket>>>;
+
+/**  DepthAI:
+ *   This is a wrapper on the host side of the device which 
+ *   helps in initializing device, creating pipeline
+ *   and fetching the frames as cv::Mat. 
+ */
+class DepthAI : public Device {
+
+public:
+    /**  Constructor:
+     *   initializing device and creating pipeline
+     *  @param usb_device (std::string): Provide the path of the device
+     *  or pass empty string to choose the default device.
+     *  @param config_file (std::string): Provides the json file which is
+     *  used by the Depthai to configure the device
+     *  @param usb2_mode (bool): set to true to connect over usb2
+     */
+    DepthAI(const std::string& usb_device, const std::string& config_file, bool usb2_mode = false);
+
+    /* API to stream output frames from device.
+     * AI models results and metadata is WIP
+     * @param output_streams (std::unordered_map<std::string, CV_mat_ptr>): output_streams is a placeholder 
+     * to extract all the streams being sent by the device. 
+     */
+    void get_frames(std::unordered_map<std::string, CV_mat_ptr>& output_streams);
```

Comment:
```suggestion
    void get_streams(std::unordered_map<std::string, CV_mat_ptr>& output_streams);
```

Reply:
Now that I understand that this function will (eventually) output all the streams (but currently only returns the cv::Mat shared pointers), we should rename this.

Reply:
couldn't we have just changed it later when we do that ?